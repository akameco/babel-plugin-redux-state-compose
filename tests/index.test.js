// @flow
import fs from 'fs'
import { join } from 'path'
import glob from 'glob'
import { transform } from 'babel-core'
import plugin from '../src'

const fixturePath = join(__dirname, 'fixtures')

for (const dir of fs.readdirSync(fixturePath)) {
  test(`snapshot ${dir}`, () => {
    const cwd = join(fixturePath, dir)
    const input =
      glob.sync(`**/App/reducer.js`, {
        cwd,
        realpath: true
      })[0] || `${cwd}/App/reducer.js`

    const filename = join(fixturePath, dir, 'state.js')
    const code = fs.readFileSync(filename, 'utf8').trim()

    const result = transform(code, {
      filename,
      babelrc: false,
      plugins: [[plugin, { input }]]
    }).code.trim()

    const separator = '\n\n      ↓ ↓ ↓ ↓ ↓ ↓\n\n'
    const formattedOutput = [code, separator, result].join('')

    expect(`\n${formattedOutput}\n`).toMatchSnapshot()
  })
}
