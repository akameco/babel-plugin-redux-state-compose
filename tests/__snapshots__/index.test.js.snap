// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot already-imported 1`] = `
"
// @flow
import type { State as AppState } from './App/reducer'

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import type { State as AppState } from './App/reducer';

export type State = {
  app: AppState;
};
"
`;

exports[`snapshot basic 1`] = `
"


      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import type { State as AppState } from \\"./App/reducer\\";

export type State = {
  app: AppState;
};
"
`;

exports[`snapshot some-state 1`] = `
"
// @flow
import type { State as HogeState } from './Hoge/reducer'

export type State = {
  hoge: HogeState,
}

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import type { State as AppState } from './App/reducer';
import type { State as HogeState } from './Hoge/reducer';

export type State = {
  app: AppState;
  hoge: HogeState;
};
"
`;

exports[`snapshot some-state2 1`] = `
"
// @flow

export type State = {
  hoge: HogeState,
}

      ↓ ↓ ↓ ↓ ↓ ↓

// @flow
import type { State as AppState } from \\"./App/reducer\\";

export type State = {
  app: AppState;
};
"
`;
